models/users.js

```diff
@@ -0,0 +1,60 @@
+var mongoose = require('mongoose');
+const dotenv = require('dotenv');
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+require('dotenv').config();
+
+var us = new mongoose.Schema({
+	name: { type: String, minlength: 6, maxlength: 125, unique: true },
+	email: { type: String, minlength: 6, maxlength: 125, unique: true },
+	password: { type: String, minlength: 6, maxlength: 125 },
+	token: { type: String, default:'' }
+});
+
+
+//saving user data
+us.pre('save', function (next) {
+	var user = this;
+	if (user.isModified('password')) {//checking if password field is available and modified
```

Comment:
Why do we generate a new hash when the password has been modified?

```suggestion
	if (user.isModified('password')) { //checking if password field is available and modified
```